### ISSUE TITLE
integ-tests: Custom synthesizer ignored during integration tests

### ISSUE THREAD

**Issue by diranged** [aws/aws-cdk#22804]
### Describe the bug

We choose to not use the default bootstrap qualifier name (`hnb659fds`), and instead rename it `default`.  For our CDK apps this works fine, as we can just update the `cdk.json` file and be done. However, we have noticed that the Integration Tests (`integ-tests-alpha` and `integ-runner` modules) do not honor either the `cdk.json` file OR passing in a custom Synthesizer option.

See Slack Thread: https://cdk-dev.slack.com/archives/C018XT6REKT/p1667598499102579

### Expected Behavior

I would expect that both the `cdk.json` and the custom synthesizer would be honored, but neither seem to be.

### Current Behavior

You seem to be stuck using the default bootstrap qualifier name, and that's a shame.

### Reproduction Steps

Example Code:
```typescript
const app = new App();
const stack = new Stack(app, "root", {
  env: env,
  synthesizer: new DefaultStackSynthesizer({
    qualifier: "default",
  }),
});
new NestedVpcStack(stack, "vpc", {});

new IntegTest(app, "IntegTest", {
  testCases: [stack],
});
```

Output:
```
✨  Synthesis time: 0.52s

IntegTest/DefaultTest/DeployAssert: building assets...

current credentials could not be used to assume 'arn:aws:iam::xxxxx:role/cdk-hnb659fds-deploy-role-xxxx-us-west-2', but are for the right account. Proceeding anyway.
root: building assets...

[0%] start: Building 38c0da6d553db6d234152b663483ada66ada1d1c171ca6453fc401569f821b28:xxxx-us-west-2
```

### CDK CLI Version

2.50.0

### Node.js Version

18.9.0

### OS

Mac OSX

### Language

Typescript

### Language Version

typescript 4.8.4

**Comment by corymhall** [comment]
@diranged `IntegTest` creates an assertions stack by default (which uses the default synthesizer). You can pass a custom stack instead. https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_integ-tests-alpha.IntegTest.html#assertionstack

```ts
const assertionStack = new Stack(app, "root", {
  env: env,
  synthesizer: new DefaultStackSynthesizer({
    qualifier: "default",
  }),
});

new IntegTest(app, "IntegTest", {
  testCases: [stack],
  assertionsStack,
});
```

**Comment by mrgrain** [comment]
Yes @diranged I agree. We should at least add this to the docs. Ideally some magic would pick up a customer synthesizer from the stack under test. But this might actually be even more confusing.

**Comment by corymhall** [comment]
@diranged thanks for the feedback! I'll leave this open to track adding better documentation. I think initially the documentation was very focused on using it within this repo, but we need to add better documentation that is focused on customers using this for their own projects.
